### DISCUSSION TITLE
indicator going up or down?

### DISCUSSION THREAD

**Discussion by bhagyashrigavande** [DaveSkender/Stock.Indicators#1199] [Help and support]
Hi there,
Is there any way by which we can identify if any indicator is going up direction or down ?
example: let's take an example of RSI . 
if current value of RSI is 60, previous is 58, previous to previous is 55 then we can say it is going in up direction i.e. RSI value is getting increasing or if current value is 60, previous value is 65, previous to previous value is 79 then it is in down trend.
Looking on chart is easy to identify if it is up trend or downtrend, how can we identify programmatically? is there any way to do this? 

I can do something as below, 
`var rsi = candleData.intraDayCandleData.GetRsi(14); `
 `var rsiLast = rsi.Last().Rsi;`
`var rsiPrev = rsi.ElementAt(rsi.Count() - 1).Rsi;`
`var rsiPrevPrev = rsi.ElementAt(rsi.Count() - 2).Rsi;`

but I think, this is not efficient way to check the trend.

**Comment by DaveSkender** [comment]
`var roc = rsi.GetRoc(1);` will give you a positive or negative change value.

ref https://dotnet.stockindicators.dev/indicators/Roc

`var slope = rsi.GetSlope(5);` is also a good choice if you want a linear regression over several periods

ref: https://dotnet.stockindicators.dev/indicators/Slope

**Reply by DaveSkender** [reply]
though a simple `bool isUp = rsiList[i] > rsiList[i-1];` works too
